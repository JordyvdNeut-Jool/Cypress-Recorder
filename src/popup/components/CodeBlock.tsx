import * as React from 'react';

export interface CodeBlockProps {
  text: string,
  destroyBlock: () => void,
  handleDragStart: () => void,
  handleDragOver: (e: React.DragEvent) => void,
  handleDrop: (e: React.DragEvent) => void,
  handleDragEnd: () => void,
}

export default ({
  text,
  destroyBlock,
  handleDragStart,
  handleDragOver,
  handleDragEnd,
  handleDrop,
}: CodeBlockProps) => {
  const i = text.indexOf('(') + 1;
  const j = text.startsWith('cy.visit')
    ? text.lastIndexOf(')')
    : text.lastIndexOf(')', text.length - 4);
  const preSelector = text.slice(0, i);
  const selector = text.slice(i, j);
  const postSelector = text.slice(j);
  return (
    <li
      className={'code-block'}
      draggable
      onDragStart={handleDragStart}
      onDragEnd={handleDragEnd}
      onDragOver={handleDragOver}
      onDrop={handleDrop}
    >
      <span>
        {preSelector}
        <mark className="selector">
          {selector}
        </mark>
        {postSelector}
      </span>
      <button type="button" className="delete" onClick={destroyBlock}>x</button>
    </li>
  );
};
