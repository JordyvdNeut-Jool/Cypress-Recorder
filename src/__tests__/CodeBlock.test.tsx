import * as React from 'react';
import { shallow } from 'enzyme';
import CodeBlock, { CodeBlockProps } from '../popup/components/CodeBlock';
import '../setupTests';

describe('CodeBlock', () => {
  let wrapper;
  const props: CodeBlockProps = {
    text: 'cy.get(\'.button\').click();',
    destroyBlock: jest.fn(),
    handleDragStart: jest.fn(),
    handleDragOver: jest.fn(),
    handleDragEnd: jest.fn(),
    handleDrop: jest.fn(),
  };
  describe('CodeBlock', () => {
    it('Should render a CodeBlock', () => {
      wrapper = shallow(<CodeBlock {...props} />);
      expect(wrapper).toMatchSnapshot();
      const item = wrapper.find('li');
      expect(item.find('mark').hasClass('selector')).toBe(true);
    });
  });
});
