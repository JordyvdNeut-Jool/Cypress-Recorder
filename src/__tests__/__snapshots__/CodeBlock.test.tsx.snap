// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock CodeBlock Should render a CodeBlock 1`] = `
<li
  className="code-block"
  draggable={true}
  onDragEnd={[MockFunction]}
  onDragOver={[MockFunction]}
  onDragStart={[MockFunction]}
  onDrop={[MockFunction]}
>
  <span>
    cy.get(
    <mark
      className="selector"
    >
      '.button'
    </mark>
    ).click();
  </span>
  <button
    className="delete"
    onClick={[MockFunction]}
    type="button"
  >
    x
  </button>
</li>
`;
